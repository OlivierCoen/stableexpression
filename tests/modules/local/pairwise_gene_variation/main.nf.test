nextflow_process {

    name "Test Process PAIRWISE_GENE_VARIATION"
    script "modules/local/pairwise_gene_variation/main.nf"
    process "PAIRWISE_GENE_VARIATION"
    tag "pairwisegenevariation"
    tag "module"

    test("Should run without failures") {

        when {
            process {
                """
                ch_counts = Channel.fromPath( '$projectDir/tests/data/merge_counts/output/all_counts.parquet', checkIfExists: true)
                input[0] = ch_counts
                """
            }
        }

        then {
            println(process.out)
            assert process.success
            assert snapshot(process.out).match()
        }

    }

}
