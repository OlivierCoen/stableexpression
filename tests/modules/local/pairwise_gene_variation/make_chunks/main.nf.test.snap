{
    "Should run without failures": {
        "content": [
            {
                "0": [
                    [
                        "count_chunk.0.parquet:md5,6b20401635b2a408a9f5271ddeb99111",
                        "count_chunk.1.parquet:md5,84f80c8f3adfcde6efff58420d68fd88",
                        "count_chunk.2.parquet:md5,503334b1bec074872581b0951937a614",
                        "count_chunk.3.parquet:md5,fefadf60ee25192be62b8f8fbe4f8f03"
                    ]
                ],
                "1": [
                    [
                        "MAKE_CHUNKS",
                        "python",
                        "3.12.8"
                    ]
                ],
                "2": [
                    [
                        "MAKE_CHUNKS",
                        "polars",
                        "1.17.1"
                    ]
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.3"
        },
        "timestamp": "2025-01-16T09:18:58.894208912"
    }
}